src/Microsoft.Health.Fhir.Api/Features/Routing/UrlResolver.cs

```diff
@@ -130,8 +130,14 @@ public Uri ResolveRouteUrl(IEnumerable<Tuple<string, string>> unsupportedSearchP
                 foreach (KeyValuePair<string, StringValues> searchParam in Request.Query)
                 {
                     // Remove the parameter if:
-                    // 1. It is the _sort parameter and the parameter is not supported for sorting
-                    // 2. The parameter is not supported.
+                    // 1. It is the _total parameter, since we only want to count for the first page of results.
+                    if (string.Equals(searchParam.Key, KnownQueryParameterNames.Total, StringComparison.OrdinalIgnoreCase) && removeTotalParameter)
```

Comment:
perf suggestion: put the removeTotalParameter first, so that we will not bother doing string.equals calls were removeTotalParameter is false.

Reply:
Thanks, implemented in fb17e80.